handlers/asset_handler.go

```diff
@@ -0,0 +1,26 @@
+package handlers
+
+import (
+	"favourites/database"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type AssetHandler struct {
+	service database.AssetService
+}
+
+func NewAssetHandler(service database.AssetService) *AssetHandler {
+	return &AssetHandler{service: service}
+}
+
+func (h *AssetHandler) GetAll(ctx *gin.Context) {
```

Comment:
I noticed in your implementation that the context is propagated end-to-end until the database. What are some advantages of doing this?

Reply:
- Timeounts: say something takes too long it can be terminated at any level, or if the request is cancelled.
- Middleware usage, say to set/check/unset cookies
- That being said also security, since in an extension cookies/tokens can be checked at any level.
- Keeping the data relative to the request for improved consistency.